src/Stratis.Bitcoin.Features.Wallet/WalletFeature.cs

```diff
@@ -183,6 +185,8 @@ public static IFullNodeBuilder UseWallet(this IFullNodeBuilder fullNodeBuilder)
                         services.AddSingleton<IBroadcasterManager, FullNodeBroadcasterManager>();
                         services.AddSingleton<BroadcasterBehavior>();
                         services.AddSingleton<WalletSettings>();
+                        services.AddSingleton<IScriptAddressReader>(new ScriptAddressReader());
```

Comment:
I believe wallet should be a new project dedicated for SmartContracts and will depend on the current wallet.